samples/quickperf/src/main/java/com/google/cloud/jdbc/quickperf/QuickPerf.java

```diff
@@ -0,0 +1,497 @@
+                            // read-only
+                            start = System.nanoTime();
+                            hasResults = statement.execute();
+                            stop = System.nanoTime() - start;
+                        } else {
+                            start = System.nanoTime();
+                            hasResults = statement.execute();
+                            connection.commit();
+                            stop = System.nanoTime() - start;
+                        }
+
+                        if (hasResults) {
+                            statement.getResultSet().close();
+                        }
+
+                        measures[i] = stop / 1000000;
+                        progress++;
+                    } catch (Exception e) {
+                        if (e.getMessage().contains("ALREADY_EXISTS:")) {
+                            System.out.println("duplicate key - retry");
+                            i--;
+                        } else {
+                            throw e;
+                        }
+                    }
+                } else if (config.getQuery().contains("INSERT") ||
+                        config.getQuery().contains("UPDATE")
+                        || config.getQuery().contains("DELETE")) {
+                    // batching
+                    try {
+                        if (config.getQueryParams() != null) {
+                            statement = parametrizeStatement(statement, config.getQueryParams());
+                        }
+
+                        statement.addBatch();
+
+                        if (batchCounter == 0 || i == config.getIterations() - 1) {
+
+                            if (isGoogleSQL) {
+                                connection.createStatement().execute(String.format("SET STATEMENT_TAG='%s'",
+                                        DEFAULT_TAG));
+                            } else {
+                                connection.createStatement().execute(String.format("SET SPANNER.STATEMENT_TAG='%s'",
+                                        DEFAULT_TAG));
+                            }
+
+                            long start = 0;
+                            long stop = 0;
+                            if (connection.getAutoCommit()) {
+                                // read-only
+                                start = System.nanoTime();
+                                int[] cnt = statement.executeBatch();
+                                stop = System.nanoTime() - start;
+                            } else {
+                                start = System.nanoTime();
+                                int[] cnt = statement.executeBatch();
+                                connection.commit();
+                                stop = System.nanoTime() - start;
+                            }
+
+                            batchCounter = config.getBatchSize();
+
+                            measures[batchRound] = stop / 1000000;
```

Comment:
I understand what's going on here, but see my comment further above as well: If you use a `Stopwatch`, then you can let this array be an array of `Duration`, which makes it clearer what it actually contains. Now it contains a float that (implicitly) contains seconds (?)